### PR TITLE
hv-overhaul: navigation examples

### PR DESCRIPTION
**Pull request by rohanbayya**

### CODE REVIEW

File: examples/navigation/back/loading/index.xml.njk [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{% from 'macros/description/index.xml.njk' import description %}
+{% from 'macros/button/index.xml.njk' import button %}
+
+{% block content %}
+  {{ description('Tapping the button below will navigate back to a different screen, with preload state') }}
```
Comment (hgray-instawork) [inline examples/navigation/back/loading/index.xml.njk:5]: maybe replace "with preload state" with something like "with a loading screen"

File: examples/navigation/close/close-loading/index.xml.njk [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{% from 'macros/description/index.xml.njk' import description %}
+{% from 'macros/button/index.xml.njk' import button %}
+
+{% block content %}
+  {{ description('Tapping the button below will navigate back to a different screen, with preload state') }}
```
Comment (hgray-instawork) [inline examples/navigation/close/close-loading/index.xml.njk:5]: Same suggestion here

File: examples/navigation/modal/index.xml.njk [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+tags: "navigation"
+hv_title: "Modal"
+hv_button_behavior: "back"
+---
+{% extends 'templates/scrollview.xml.njk' %}
+{% from 'macros/description/index.xml.njk' import description %}
+{% from 'macros/button/index.xml.njk' import button %}
+
+{% block content %}
+  {{ description('Open a new modal screen!') }}
```
Comment (hgray-instawork) [inline examples/navigation/modal/index.xml.njk:12]: silly, but maybe we don't need the exclamation?

File: examples/navigation/navigate/index.xml.njk [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+{% from 'macros/button/index.xml.njk' import button %}
+
+{% block content %}
+  {{ description('Tapping the button below will request an XML screen and push it onto the stack.') }}
+
+  {% call button('Navigate to home') -%}
+    <behavior action="navigate" href="/index.xml" />
+  {%- endcall %}
+
+  {% call button('Navigate with indicator') -%}
```
Comment (hgray-instawork) [inline examples/navigation/navigate/index.xml.njk:18]: indicator or screen?

File: examples/navigation/new/index.xml.njk [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+{% endblock %}
+
+{% block content %}
+  {{ description('Tapping the button below will request an XML screen and show it in a new modal.') }}
+
+  {% call button('New') -%}
+    <behavior action="new" href="/navigation/modal.xml" />
+  {%- endcall %}
+
+  {% call button('New with indicator') -%}
```
Comment (hgray-instawork) [inline examples/navigation/new/index.xml.njk:40]: indicator vs. screen?

File: examples/navigation/push/index.xml.njk [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+{% endblock %}
+
+{% block content %}
+  {{ description('Tapping the button below will request an XML screen and push it onto the stack.') }}
+
+  {% call button('Push') -%}
+    <behavior action="push" href="/navigation/pushed.xml" />
+  {%- endcall %}
+
+  {% call button('Push with indicator') -%}
```
Comment (hgray-instawork) [inline examples/navigation/push/index.xml.njk:40]: indicator vs. screen?

**Review by hgray-instawork — APPROVED** [review]

File: examples/navigation/new/index.xml.njk (hunk 5 above)
Comment (rohanbayya) [reply examples/navigation/new/index.xml.njk:40]: this demonstrates opening a new modal with a loading indicator. (not screen)

File: examples/navigation/push/index.xml.njk (hunk 6 above)
Comment (rohanbayya) [reply examples/navigation/push/index.xml.njk:40]: this demonstrates pushing a screen with a loading indicator

**Review by hgray-instawork — APPROVED** [review]

**Pull request merged**